Refactor: clean codes and fix typo at comments

aeharvlee opened the pull request:
* There was already defined function (= `crypto.Keccak256`) doing `keccak256`.
* It is more maintainable to use already defined function.
* It passed the test already defined in [account_test.go](https://github.com/ethereum/go-ethereum/blob/master/accounts/accounts_test.go)

holiman reviewed (commented):
I think having three lines instead of one might be preferrable over a package dependency. Especially since the `crypto` package is, I think, C-heavy.

aeharvlee commented:
@holiman 
I did bench tests for this.

**Source Code** in `accounts/accounts.go`
```go
func TextAndHash(data []byte) ([]byte, string) {
    msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(data), string(data))
    return crypto.Keccak256([]byte(msg)), msg
}

func LegacyTextAndHash(data []byte) ([]byte, string) {
    msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(data), string(data))
    hasher := sha3.NewLegacyKeccak256()
    hasher.Write([]byte(msg))
    return hasher.Sum(nil), msg
}
```

Added **Test Code** in `accounts/accounts_test.go`
```go
func BenchmarkLegacyTextAndHash(b *testing.B) {
    for n := 0; n < b.N; n++ {
        LegacyTextAndHash([]byte("Hello Joe"))
    }
}

func BenchmarkTextAndHash(b *testing.B) {
    for n := 0; n < b.N; n++ {
        TextAndHash([]byte("Hello Joe"))
    }
}
```

And the result is like below.
Refactored codes is more faster than before.

```shell
go test -bench=.                        
goos: darwin
goarch: amd64
pkg: github.com/ethereum/go-ethereum/accounts
cpu: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
BenchmarkLegacyTextAndHash-16            1265701               927.8 ns/op
BenchmarkTextAndHash-16                  1508013               797.6 ns/op
PASS
ok      github.com/ethereum/go-ethereum/accounts        4.268s
```

fjl commented:
It's not about the speed. What @holiman says is, we are use package sha3 instead of package crypto here because it avoids the dependency on libsecp256k1.

aeharvlee commented:
@fjl Thanks for your reply.
Understood but why avoiding dependency on libsecp256k1 is important? Does it related with performance or something?

fjl commented:
It is not related to performance. The way we think about it is like this: if you are building an application, you want to pull in only the libraries you really need. For the abstract functionality of package accounts, the low-level crypto library is not required, so we don't want to depend on it here.

There is a saying in the Go community: `a little copying is better than a little dependency`. I think it applies here.

aeharvlee commented:
@fjl 
Jeez... Really appreciate your detailed explaination. Really impressed. Thanks.
